### ISSUE TITLE
[Search v1] - Tax amount for distance expense in Search and transaction thread is different

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43838]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.84-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A 
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- Track tax is enabled
- The distance rate to be used has a non-zero tax rate and "Tax reclaimable on" amount is lower than the distance rate
1. Go to staging.new.expensify.com
2. Go to workspace chat
3. Submit a distance expense using the distance rate from the precondition
4. Go to transaction thread
5. Note the tax amount
6. Go to Search
7. Note that tax amount for the same distance expense in Step 4

## Expected Result:
The tax amount in Search page is the same as the tax amount in transaction thread

## Actual Result:
The tax amount in Search page is different compared to the tax amount in transaction thread

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/f672b70e-22a2-4e1c-a3f7-1092c5ac4bbc

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@laurenreidexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by laurenreidexpensify** [comment]
@anmurali handing over to you in prep for my parental leave - haven't had a chance to review yet but this looks vip-vsb

**Comment by anmurali** [comment]
Pulled it internal since it is a BE issue.

**Comment by trjExpensify** [comment]
Ah, interesting! @luacmartins and @MonilBhavsar, looks like the `taxAmount` field on the search page needs to maybe account for different sources when it comes to distance expenses.

**Comment by luacmartins** [comment]
Agreed, seems like we save tax amount for distance requests in a different place. @MonilBhavsar do you know where that's saved?

**Comment by MonilBhavsar** [comment]
Interesting, it should be returned correctly in `taxAmount` field in transactions.
Though, there was a bug where taxAmount was not being returned correctly for distance requests and was fix and deployed last week https://github.com/Expensify/Auth/pull/11410.

Can we retry and see if this is still reproducible?

**Comment by anmurali** [comment]
I am not sure how to turn on those tax fields at the distance request level. This is all I see:
<img width="1433" alt="image" src="https://github.com/Expensify/App/assets/14225673/899ad826-24e7-4874-817e-21a742aa1bb8">

**Comment by trjExpensify** [comment]
Ah, you need to enable `Track Tax` on Distance first. Click the `Settings` button on the Distance rates page:

<img width="1509" alt="image" src="https://github.com/Expensify/App/assets/16232057/e88a5e0d-44e0-4d72-956e-7f54c1172155">

<img width="1511" alt="image" src="https://github.com/Expensify/App/assets/16232057/e96131f7-1745-4ffc-856c-041e7edb9c1b">

**Comment by trjExpensify** [comment]
Is it still reproducible?

**Comment by anmurali** [comment]
![image](https://github.com/user-attachments/assets/ddb219d7-1f04-48e1-8ced-afa3ec2583d0)

![image](https://github.com/user-attachments/assets/77e6f25e-a932-4594-b457-7ad516e55fc2)

**Comment by anmurali** [comment]
@MonilBhavsar - I can reproduce. See screenshots above. Should this be internal or external?

**Comment by trjExpensify** [comment]
Going to move this into #wave-control for search.

**Comment by anmurali** [comment]
Waiting for someone to pick this up, already upgraded to a hot pick.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by IuliiaHerets** [comment]
Issue is still reproduce, build - v9.0.48-0

https://github.com/user-attachments/assets/2900c715-f852-41c7-b03e-964cd6482343

**Comment by anmurali** [comment]
@dylanexpensify - can you add this to your list of hot picks to track?

**Comment by anmurali** [comment]
Pinged #expense curator in the channel [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1730857640463929)

**Comment by luacmartins** [comment]
@MonilBhavsar are you able to look into this one?

**Comment by MonilBhavsar** [comment]
Got a PR up locally but moved to daily issues. Will shift to this

**Comment by MonilBhavsar** [comment]
I've got a draft PR up. ETA is EOW

**Comment by MonilBhavsar** [comment]
Submitted PR for review

**Comment by twisterdotcom** [comment]
I am removing hot pick from any issues with an internal engineer assigned so other engineers can know it's available.

**Comment by MonilBhavsar** [comment]
The fix was deployed. Closing
